### PR TITLE
[itkDataImage] save&load metaData on export&import

### PR DESCRIPTION
**Pull request by mathildemerle**
Save study and patient names, modality (which needed to be different of "Modality") and patient ID.

### CODE REVIEW

**Comment by fcollot** [comment]
Why not all metadata ?

**Comment by mathildemerle** [comment]
My first problem was that i could not access to medUtilities to reach medUtilities::metaDataKeysToCopyForDerivedData(), Loic tried also a bit without success. Without that, a loop on medData->metaDataList() gives way too much of information, and the exported data crashes when it's imported. 
Concerning the loop on metaDataKeysToCopyForDerivedData, even if i could access to it, there is a problem with names as seen in "Modality" which as far as i saw, overwrite a needed header of the file. I think.
I can test every metaData from medMetaDataKeys after, in order to see for each if i can keep the original name or not, but loic or you (i do not remember ^^') told me he needed this PR.

**Comment by fcollot** [comment]
Ok I see.
Maybe just factorise this code with a loop over a string list containing the keys ?

**Comment by fcollot** [comment]
(from medUtilities)
We should be keeping the following metadata:

    keys << medMetaDataKeys::PatientID.key()
         << medMetaDataKeys::PatientName.key()
         << medMetaDataKeys::Age.key()
         << medMetaDataKeys::BirthDate.key()
         << medMetaDataKeys::Gender.key()
         << medMetaDataKeys::Description.key()
         << medMetaDataKeys::StudyID.key()
         << medMetaDataKeys::StudyDicomID.key()
         << medMetaDataKeys::StudyDescription.key()
         << medMetaDataKeys::Institution.key()
         << medMetaDataKeys::Referee.key()
         << medMetaDataKeys::StudyDate.key()
         << medMetaDataKeys::StudyTime.key()
         << medMetaDataKeys::Modality.key()
         << medMetaDataKeys::Performer.key()
         << medMetaDataKeys::Report.key()
         << medMetaDataKeys::Protocol.key()
         << medMetaDataKeys::Orientation.key()
         << medMetaDataKeys::Origin.key()
         << medMetaDataKeys::AcquisitionDate.key()
         << medMetaDataKeys::AcquisitionTime.key();

**Comment by mathildemerle** [comment]
Yep, i was on that. So, i needed to test each of these medMetaDataKeys, it's done. There are only 2 of them which needed to be saved in the output file with an other name.

File: src/medImageIO/itkDataImageReaderBase.cpp [hunk 1]
```diff
@@ -242,21 +242,79 @@ bool itkDataImageReaderBase::read_image(const QString& path,const char* type)
...
+        medData->setMetaData(medMetaDataKeys::Modality.key(), QString(value.c_str()));
+    }
+    if (itk::ExposeMetaData(dict, "MED_ORIENTATION", value))
+    {
+        medData->setMetaData(medMetaDataKeys::Orientation.key(), QString(value.c_str()));
+    }
     return true;
 }
 
+QStringList itkDataImageReaderBase::metaDataKeysToCopyForDerivedData()
```
Comment (fcollot) [inline src/medImageIO/itkDataImageReaderBase.cpp]: technically we are not setting the metadata of derived data in this code. ```metaDataKeysToCopy```is sufficient here

**Comment by LoicCadour** [comment]
Works well (the gender is not kept though).
I agree with Flo's comment.

**Comment by mathildemerle** [comment]
I've updated the function name :)

**Comment by fcollot** [comment]
ok

**Pull request merged**